apps/api/src/types/index.ts

```diff
@@ -1196,3 +1196,32 @@ export const BucketParamSchema = z.coerce
   .default(30);
 
 export type BucketParam = z.infer<typeof BucketParamSchema>;
+
+export const TimestampSchema = z.coerce.date();
+
+export const SnapshotSchema = z.object({
+  id: z.number(),
+  competitionId: UuidSchema,
+  agentId: UuidSchema,
+  timestamp: TimestampSchema,
+  totalValue: z.number(),
+});
+
+/**
+ * Snakecase version of the snapshot schema, this is convenient for parsing raw
+ * query results.
+ */
+export const SnapshotDbSchema = z.object({
+  id: z.number(),
+  competition_id: UuidSchema,
```

Comment:
for consistency, it there an easy way to make these all camel case?

Reply:
I'm not sure what the best thing to do here is, but the db.execute( with the `sql` template literal means the queries return snake case.